src/sentry/static/sentry/app/views/alerts/details/header.tsx

```diff
@@ -69,89 +66,123 @@ export default class DetailsHeader extends React.Component<Props> {
       onSubscriptionChange,
     } = this.props;
     const isIncidentReady = !!incident && !hasIncidentDetailsError;
-    const dateStarted = incident && moment(incident.dateStarted).format('LL');
+    const dateStarted = incident && moment(new Date(incident.dateStarted)).format('llll');
```

Comment:
There's a `use24Hours()` util func that gets the user's preference, we may want to use that to determine the date format.